soc/st/stm32/stm32f4/Kconfig.soc

```diff
@@ -0,0 +1,113 @@
+# STMicroelectronics STM32F4 MCU line
+
+# Copyright (c) 2016 Linaro Limited.
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_SERIES_STM32F4X
+	bool
+	select SOC_FAMILY_STM32
+
+config SOC_SERIES
+	default "stm32f4" if SOC_SERIES_STM32F4X
+
+config SOC_STM32F401XC
+	bool
+	select SOC_SERIES_STM32F4X
+
+config SOC_STM32F401XE
+	bool
+	select SOC_SERIES_STM32F4X
+
+config SOC_STM32F405XG
+	bool
+	select SOC_SERIES_STM32F4X
+
+config SOC_STM32F407XE
+	bool
+	select SOC_SERIES_STM32F4X
+
+config SOC_STM32F407XG
+	bool
+	select SOC_SERIES_STM32F4X
+
+config SOC_STM32F410RX
+	bool
+	select SOC_SERIES_STM32F4X
+
+config SOC_STM32F411XE
+	bool
+	select SOC_SERIES_STM32F4X
+
+config SOC_STM32F412CX
+	bool
+	select SOC_SERIES_STM32F4X
+
+config SOC_STM32F412RX
+	bool
+	select SOC_SERIES_STM32F4X
+
+config SOC_STM32F412VX
+	bool
+	select SOC_SERIES_STM32F4X
+
+config SOC_STM32F412ZX
+	bool
+	select SOC_SERIES_STM32F4X
+
+config SOC_STM32F413XX
+	bool
+	select SOC_SERIES_STM32F4X
+
+config SOC_STM32F415XX
+	bool
+	select SOC_SERIES_STM32F4X
+
+config SOC_STM32F417XX
+	bool
+	select SOC_SERIES_STM32F4X
+
+config SOC_STM32F423XX
+	bool
+	select SOC_SERIES_STM32F4X
+
+config SOC_STM32F427XX
+	bool
+	select SOC_SERIES_STM32F4X
+
+config SOC_STM32F429XX
+	bool
+	select SOC_SERIES_STM32F4X
+
+config SOC_STM32F437XX
+	bool
+	select SOC_SERIES_STM32F4X
+
+config SOC_STM32F446XX
+	bool
+	select SOC_SERIES_STM32F4X
+
+config SOC_STM32F469XX
+	bool
+	select SOC_SERIES_STM32F4X
+
+config SOC
+	default "stm32f401xc" if SOC_STM32F401XC
+	default "stm32f401xe" if SOC_STM32F401XE
+	default "stm32f405xx" if SOC_STM32F405XG
```

Comment:
Let's take the opportunity to fix it as SOC_STM32F405XX.
To be done as a follow up commit to make it visible